azurerm/resource_arm_container_registry.go

```diff
@@ -231,9 +302,16 @@ func resourceArmContainerRegistryUpdate(d *schema.ResourceData, meta interface{}
 	oldGeoReplicationLocations := old.(*schema.Set)
 	newGeoReplicationLocations := new.(*schema.Set)
 
+	networkRuleSet := expandNetworkRuleSet(d)
+	//NetworkRuleSet is only supported by Premium SDK
+	if networkRuleSet != nil && !strings.EqualFold(sku, string(containerregistry.Premium)) {
+		return fmt.Errorf("`network_rule_set` can only be specified for a Premium Sku.")
+	}
+
 	parameters := containerregistry.RegistryUpdateParameters{
 		RegistryPropertiesUpdateParameters: &containerregistry.RegistryPropertiesUpdateParameters{
 			AdminUserEnabled: utils.Bool(adminUserEnabled),
+			NetworkRuleSet:   networkRuleSet,
```

Comment:
I think we are going to need to conditionally set this property:
```
------- Stdout: -------
=== RUN   TestAccAzureRMContainerRegistry_geoReplication
=== PAUSE TestAccAzureRMContainerRegistry_geoReplication
=== CONT  TestAccAzureRMContainerRegistry_geoReplication
--- FAIL: TestAccAzureRMContainerRegistry_geoReplication (463.91s)
    testing.go:568: Step 4 error: errors during apply:
        
        Error: `network_rule_set` can only be specified for a Premium Sku.
        
          on /opt/teamcity-agent/temp/buildTmp/tf-test896584085/main.tf line 7:
          (source code not available)
        
        
FAIL
```

Reply:
@katbyte 
Is this happening because the test does not remove the `network_rule_set` and then a subsequent test tried to set the SKU to `Basic`? So the solution would be to remove the `network_rule_set` after the test has ran?

Reply:
yes, might have to specify a empty value there to make sure its not set due to it being computed

Reply:
Have taken this approach